sale_coupon_domain_product_reward_fixed_price/models/sale_order.py

```diff
@@ -0,0 +1,21 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _set_reward_fixed_price_for_lines(self, program):
+        """
+        Update discount field by program
+        """
+        price_unit = program.price_unit
```

Comment:
For what you create separate variable and use use this variable only in condition and only once.